File: app/src/main/java/com/example/android/popularmovies/data/Contract/MovieContract.java
package com.example.android.popularmovies.data.Contract;

import android.net.Uri;
import android.provider.BaseColumns;

/**
 * Created by msk-1196 on 7/13/17.
 */

public class MovieContract extends BaseContract {
    public static final String PATH_MOVIE = "movie";

    public static final class MovieEntry implements BaseColumns{
        public static final Uri CONTENT_URI =
                BASE_CONTENT_URI.buildUpon().appendPath(PATH_MOVIE).build();

        public static final String TABLE_NAME = "movie";

        public static final String COLUMN_MOVIEDB_ID = "moviedb_id";
        public static final String COLUMN_TITLE = "title";
        public static final String COLUMN_OVERVIEW = "overview";
        public static final String COLUMN_BACKDROP = "backdrop";
        public static final String COLUMN_POSTER = "poster";
        public static final String COLUMN_VOTE_AVG = "vote_avg";
        public static final String COLUMN_VOTE_COUNT = "vote_count";
        public static final String COLUMN_POPULAITY = "popularity";
        public static final String COLUMN_RELEASE_DATE = "release_date";
        public static final String COLUMN_IS_FAVORITE = "is_favorite";


        public static Uri buildMovieWithMovieDBId(long movieDBId){
            return CONTENT_URI.buildUpon().appendPath(Long.toString(movieDBId)).build();
        }

        public static Uri buildTrailerUri(long movieDBId){
            return CONTENT_URI.buildUpon().appendPath(Long.toString(movieDBId))
                    .appendPath(TrailerContract.PATH_TRAILER).build();
        }

        public static Uri buildReviewrUri(long movieDBId){
            return CONTENT_URI.buildUpon().appendPath(Long.toString(movieDBId))
                    .appendPath(ReviewContract.PATH_REVIEW).build();
        }


    }
}

File: app/src/main/java/com/example/android/popularmovies/data/Contract/MovieContract.kt
package com.example.android.popularmovies.data.Contract

import android.net.Uri
import android.provider.BaseColumns

/**
 * Created by msk-1196 on 7/13/17.
 */

class MovieContract : BaseContract() {

    class MovieEntry {
        companion object {
            val CONTENT_URI = BASE_CONTENT_URI.buildUpon().appendPath(PATH_MOVIE).build()

            val TABLE_NAME = "movie"

            val _ID = "_id"
            val COLUMN_MOVIEDB_ID = "moviedb_id"
            val COLUMN_TITLE = "title"
            val COLUMN_OVERVIEW = "overview"
            val COLUMN_BACKDROP = "backdrop"
            val COLUMN_POSTER = "poster"
            val COLUMN_VOTE_AVG = "vote_avg"
            val COLUMN_VOTE_COUNT = "vote_count"
            val COLUMN_POPULAITY = "popularity"
            val COLUMN_RELEASE_DATE = "release_date"
            val COLUMN_IS_FAVORITE = "is_favorite"


            fun buildMovieWithMovieDBId(movieDBId: Long): Uri {
                return CONTENT_URI.buildUpon().appendPath(java.lang.Long.toString(movieDBId)).build()
            }

            fun buildTrailerUri(movieDBId: Long): Uri {
                return CONTENT_URI.buildUpon().appendPath(java.lang.Long.toString(movieDBId))
                        .appendPath(TrailerContract.PATH_TRAILER).build()
            }

            fun buildReviewrUri(movieDBId: Long): Uri {
                return CONTENT_URI.buildUpon().appendPath(java.lang.Long.toString(movieDBId))
                        .appendPath(ReviewContract.PATH_REVIEW).build()
            }
        }


    }

    companion object {
        const val PATH_MOVIE = "movie"
    }
}
